File: 0038-count-and-say/0038-count-and-say.java
class Solution {
    public String countAndSay(int n) {
        if(n == 0) return "";
        String ans = "1";

        while(--n > 0) {
            StringBuilder curr = new StringBuilder();
            for(int i = 0; i<ans.length(); i++) {
                int count = 1;
                while(i + 1 < ans.length() && ans.charAt(i) == ans.charAt(i + 1)) {
                    count++;
                    i++;
                }
                curr.append(count).append(ans.charAt(i));
            }

            ans = curr.toString();
        }

        return ans;
    }
}

File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n == 0) return "";
        string ans = "1";
        
        while(--n){
            string curr = "";
            for(int i = 0; i<ans.size(); i++){
                int count = 1;
                while(i+1 < ans.size() && ans[i] == ans[i+1]){
                    count++, i++;
                }
                curr += to_string(count) + ans[i];
            }
            ans = curr;
        }
        
        return ans;
    }
};